### PR TITLE
fill host-placeholder before clearing cache

### LINKED ISSUE #2531: Clearing cache does not work if host placeholder in webspace.xml is used
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | Specific version or SHA of a commit
| Browser Version | Browser name and version

#### Actual Behavior

If the `{host}` placeholder in the webspace config is used, clearing the cache does not work anymore, because the placeholder is passed through till https://github.com/sulu/sulu/blob/develop/src/Sulu/Component/HttpCache/Handler/PathsHandler.php#L95

#### Expected Behavior

The `{host}'  placeholder must be replaced somewhere, so that an actual URL is the result. Otherwise the cache won't be purged.

#### Steps to Reproduce

Has to be done in prod environment.

1. Create a webspace.xml with the `{host}` parameter
2. Create a page and save it
3. Load the page in the browser
4. Reload it again an check the response time to make sure it is cached
5. Save the page in the Admin UI
6. Reload it again and see that the cache hasn't been purged

#### Possible Solutions

Replace `{host}` in the URL with the current host.

### PR DESCRIPTION
**Pull request by nnatter**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2531
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

This PR (partially) fixes cache-clearing on page-publishing when using a host-placeholder in webspace.xml.

The misbehaviour is fixed by storing the host of the request on a publish request and fill the host-placeholder of the url with the stored host when clearing cache.
This method does not fix the problem on multi-host portals.

### CODE REVIEW

File: src/Sulu/Component/HttpCache/Handler/PathsHandler.php [hunk 1]
```diff
@@ -35,28 +37,45 @@ class PathsHandler implements HandlerFlushInterface, HandlerInvalidateStructureI
...
      */
     private $structuresToInvalidate = [];
 
     /**
      * @param WebspaceManagerInterface $webspaceManager
      * @param ProxyClientInterface     $proxyClient
-     * @param string                   $environment     - kernel envionment, dev, prod, etc.
+     * @param RequestStack             $requestStack
+     * @param ReplacerInterface        $replacer
+     * @param string                   $environment     - kernel envionment, dev, prod, etc
```
Comment (danrot) [inline src/Sulu/Component/HttpCache/Handler/PathsHandler.php]: We don't align the text like that, simple write e.g.
```
@param RequestStack $requestStack
```

File: src/Sulu/Component/HttpCache/Tests/Unit/Handler/PathsHandlerTest.php [hunk 2]
```diff
@@ -51,10 +56,18 @@ public function setUp()
 
         $this->webspaceManager = $this->prophesize('Sulu\Component\Webspace\Manager\WebspaceManagerInterface');
         $this->proxyClient = $this->prophesize('FOS\HttpCache\ProxyClient\Invalidation\PurgeInterface');
+        $this->requestStack = $this->prophesize('Symfony\Component\HttpFoundation\RequestStack');
+        $this->request = $this->prophesize('Symfony\Component\HttpFoundation\Request');
+        $this->replacer = $this->prophesize('Sulu\Component\Webspace\Url\ReplacerInterface');
```
Comment (nnatter) [inline src/Sulu/Component/HttpCache/Tests/Unit/Handler/PathsHandlerTest.php]: class property

File: src/Sulu/Bundle/HttpCacheBundle/Tests/Unit/DependencyInjection/SuluHttpCacheExtensionTest.php [hunk 3]
```diff
@@ -22,6 +22,8 @@ public function setUp()
         $this->container->setParameter('kernel.environment', 'test');
         $this->container->set('sulu_core.webspace.webspace_manager', $this->getMock('Sulu\Component\Webspace\Manager\WebspaceManagerInterface'));
         $this->container->set('sulu.content.type_manager', $this->getMock('Sulu\Component\Content\ContentTypeManagerInterface'));
+        $this->container->set('request_stack', $this->getMock('Symfony\Component\HttpFoundation\RequestStack'));
+        $this->container->set('sulu_core.webspace.webspace_manager.url_replacer', $this->getMock('Sulu\Component\Webspace\Url\Replacer'));
```
Comment (nnatter) [inline src/Sulu/Bundle/HttpCacheBundle/Tests/Unit/DependencyInjection/SuluHttpCacheExtensionTest.php]: deprecated

Comment (danrot) [inline src/Sulu/Bundle/HttpCacheBundle/Tests/Unit/DependencyInjection/SuluHttpCacheExtensionTest.php]: And I just saw that the `container` variable is not initialized, please fix that also 😃

Comment (nnatter) [inline src/Sulu/Bundle/HttpCacheBundle/Tests/Unit/DependencyInjection/SuluHttpCacheExtensionTest.php]: i think the `container`-variable is initialized in the `setUp`-method `AbstractContainerBuilderTestCase`-class

**Pull request merged**